### PR TITLE
OGR data provider: use transactions when changing attributes 

### PR DESCRIPTION
**Pull request by landam**
See http://hub.qgis.org/issues/16216

### CODE REVIEW

**Comment by rouault** [comment]
The PR should target master (with potential backport in 2.18)

File: src/providers/ogr/qgsogrprovider.cpp [hunk 1]
```diff
@@ -1524,6 +1524,8 @@ bool QgsOgrProvider::renameAttributes( const QgsFieldNameMap& renamedAttributes
 
 bool QgsOgrProvider::changeAttributeValues( const QgsChangedAttributesMap &attr_map )
 {
+  bool hasTransaction;
```
Comment (nyalldawson) [inline src/providers/ogr/qgsogrprovider.cpp]: Can you move this declaration down to line 1539 when the value is assigned? Then we don't have an uninitisalised variable floating around.

**Review by nyalldawson — COMMENTED** [review]
Nice performance boost... This will certainly be appreciated!

File: src/providers/ogr/qgsogrprovider.cpp (hunk 1 above)
Comment (landam) [inline src/providers/ogr/qgsogrprovider.cpp]: Done in c391341

**Comment by landam** [comment]
Pull request against master: https://github.com/qgis/QGIS/pull/4175

**Comment by m-kuhn** [comment]
Closing this for now.

Please reopen a new backported PR when the issues on the master branch are ironed out. Thanks, looking forward to this!

**Pull request closed without merging**